sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateOrdering.scala

```diff
@@ -154,12 +154,14 @@ class LazilyGeneratedOrdering(val ordering: Seq[SortOrder]) extends Ordering[Int
   private[this] var generatedOrdering = GenerateOrdering.generate(ordering)
 
   def compare(a: InternalRow, b: InternalRow): Int = {
+    if (generatedOrdering == null) {
```

Comment:
Should making the generatedOrdering as `lazy val` be enough? 

see http://fdahms.com/2015/10/14/scala-and-the-transient-lazy-val-pattern/